ci_framework/roles/devscripts/tasks/sub_tasks/15_network.yml

```diff
@@ -0,0 +1,127 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Gather external interface name.
+  ansible.builtin.set_fact:
+    iface_name: "{{ ansible_default_ipv4.interface }}"
+
+- name: Enable local DNS resolution.
+  become: true
+  ansible.builtin.copy:
+    dest: "/etc/NetworkManager/conf.d/00-use-dnsmasq.conf"
+    owner: root
+    group: root
+    mode: '0644'
+    content: |
+      [main]
+      dns = dnsmasq
+
+- name: Add resolution using hosts file.
+  become: true
+  ansible.builtin.copy:
+    dest: "/etc/NetworkManager/dnsmasq.d/98-local-hosts.conf"
+    content: |
+      addn-hosts=/etc/hosts
+
+- name: Enabling forwarding to external DNS servers.
+  become: true
+  ansible.builtin.copy:
+    dest: "/etc/NetworkManager/dnsmasq.d/99-forwarders.conf"
+    content: "server={{ ansible_dns.nameservers | join('\nserver=') }}"
+
+- name: Enabling local resolution of OCP machine network.
+  become: true
+  when: cifmw_devscripts_dns_entries is defined
+  vars:
+    ocp_domain_name: "{{ cifmw_devscripts_config['cluster_name'] }}.{{ cifmw_devscripts_config['base_domain'] }}"
+    ocp_api_address: "{{ cifmw_devscripts_dns_entries['api'] }}"
+    ocp_ingress_address: "{{ cifmw_devscripts_dns_entries['ingress'] }}"
+  ansible.builtin.template:
+    src: "templates/01-openshift.conf.j2"
+    dest: "/etc/NetworkManager/dnsmasq.d/01-openshift.conf"
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Adding listener to the external bridge.
+  vars:
+    listen_addresses: "127.0.0.1,{{ ansible_default_ipv4.address}}"
+  block:
+    - name: Collect all the extra network IP addresses
+      when: cifmw_devscripts_extra_net_infra | length > 0
+      vars:
+        addr: "{{ item | split('/') | first }}"
```

Comment:
Here you're feeding key/value. This is failing molecule test.
> * `cifmw_devscripts_extra_net_infra` (list) A list of extra networks to be created. Networks is a key/value

It probably needs to be `{{ item.ipv4_address | split('/') | first }}`

Reply:
Thank you. 

It should be either `item.ipv4_address or item.ipv6_address`.